### PR TITLE
trip's week3 assessment

### PR DESCRIPTION
**Pull request by jlmooreiii**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -2,7 +2,44 @@
...
+    //Check to make sure the uer put in a number as an arg
+    if (typeof num !== "number") {
+        return "Please use a postitive number as the argument for the function fibonacci(num)."
+    } else {
+        //Make sure the user put in a positive whole number as an arg.
+        if (num <= 0 && num % 1 !== 0) {
+            return "Please use a postitive number as the argument for the function fibonacci(num).";
+        //If the user only wants the first number, just give it to them because otherwise my logic will break
+        } else if (num == 1) {
+            return [1];
```
Comment (andeeisaacs) [inline code-challenges.js:18]: 🤣  This is great.

File: code-challenges.js [hunk 2]
```diff
@@ -2,7 +2,44 @@
...
 
 // --------------------1) Create a function that returns the first 10 numbers of the Fibonacci sequence in an array. Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
-
+//Create a function that give as many numbers of the Fibonacci sequence as an array
+const fibonacci = (num) => {
+    //Declare an array with the first two numbers of the Fib. sequence
+    let fibArray = [1, 1];
+    //Check to make sure the uer put in a number as an arg
+    if (typeof num !== "number") {
+        return "Please use a postitive number as the argument for the function fibonacci(num)."
```
Comment (andeeisaacs) [inline code-challenges.js:11]: I love that you're checking for the number. Great idea.

File: code-challenges.js [hunk 3]
```diff
@@ -2,7 +2,44 @@
...
+              }
+
+    }
+    //return the array 
+    return fibArray;
+}
+//Check to make sure the function works.
+console.log("-----CHALLENGE 1-----");
+console.log("This is the result of my function fibonacci(10):");
+console.log(fibonacci(10));
```
Comment (andeeisaacs) [inline code-challenges.js:39]: What a BEAUT. Nice work!!!

File: code-challenges.js [hunk 4]
```diff
@@ -13,7 +50,29 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
...
+console.log("This is the result of my function numFilter(fullArr1):");
+console.log(numFilter(fullArr1));
+console.log("The expected result is:");
+console.log("[-9, 9, 7, 199]");
+console.log("\n");
+console.log("This is the result of my function numFilter(fullArr2):");
+console.log(numFilter(fullArr1));
+console.log("The expected result is:");
+console.log("[-9, 9, 7, 199]");
+console.log("\n");
```
Comment (andeeisaacs) [inline code-challenges.js:75]: YES! Nice work!!!

File: code-challenges.js [hunk 5]
```diff
@@ -23,13 +82,90 @@ var middleLetters1 = "hello"
...
+console.log("This is the result of my function getMiddleChar(middleLetters1):");
+console.log(getMiddleChar(middleLetters1));
+console.log("The expected result is:");
+console.log("l");
+console.log("\n");
+console.log("This is the result of my function getMiddleChar(middleLetters2):");
+console.log(getMiddleChar(middleLetters2));
+console.log("The expected result is:");
+console.log("oc");
+console.log("\n");
```
Comment (andeeisaacs) [inline code-challenges.js:129]: Beautiful consideration of the edge cases. I love it!

File: code-challenges.js [hunk 6]
```diff
@@ -23,13 +82,90 @@ var middleLetters1 = "hello"
...
+class Sphere {
+    //Give the class a constructor that takes the argument of radius and units
+    constructor(radius, units = "units") {
+        this.radius = radius;
+        this.units = units;
+    }
+    //Give the class a method that canluates its volume and returns it
+    getArea() {
+        let area = 4*(3.1415)*(this.radius)*(this.radius);
+        return ` ${area} ${this.units} cubed`;
```
Comment (andeeisaacs) [inline code-challenges.js:146]: Great work here. 

Another way we could do this, is with some built in methods such as Math.PI and Math.Pow. If you're interested in learning more about these, take a look here: 
Math.Pow: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow
Math.PI: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/PI

File: code-challenges.js [hunk 7]
```diff
@@ -23,13 +82,90 @@ var middleLetters1 = "hello"
...
+console.log("mySphere2:"); 
+console.log(mySphere2); 
+console.log("\n");
+console.log("mySphere3:"); 
+console.log(mySphere3); 
+console.log("\n");
+console.log(`The expected area of myShere1 is 201.056 units squared. The output area is ${mySphere1.getArea()}.`);
+console.log(`The expected area of myShere2 is 452.376 inches squared. The output area is ${mySphere2.getArea()}.`);
+console.log(`The expected area of myShere3 is 804.224 cm squared. The output area is ${mySphere3.getArea()}.`);
+console.log("\n");
```
Comment (andeeisaacs) [inline code-challenges.js:168]: NICE! 👍  🎖️

File: code-challenges.js [hunk 8]
```diff
@@ -39,3 +175,40 @@ var numbersToAdd2 = [0, 7, -8, 12]
...
+console.log("This is the result of my function accumulatingSum(numbersToAdd2):");
+console.log(accumulatingSum(numbersToAdd2));
+console.log("The expected result is:");
+console.log("[0, 7, -1, 11]");
+console.log("\n");
+console.log("This is the result of my function accumulatingSum(numbersToAdd3):");
+console.log(accumulatingSum(numbersToAdd3));
+console.log("The expected result is:");
+console.log("[]");
+console.log("\n");
```
Comment (andeeisaacs) [inline code-challenges.js:214]: Heckin' yes.  KILLED IT. 🤠

File: interview-questions.md [hunk 9]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is a technique that dries up code by allowing the developer to quickly access key value pairs inside objects that are nested deeply in objects. It allowds you to assign the value of the key value pair to its address, similar to a variable call, so that you can call upon the specified value multiple times without having to type the whole dot notation addess of the value every time.
```
Comment (andeeisaacs) [inline interview-questions.md:7]: 👏  Very good.

File: interview-questions.md [hunk 10]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer: I fount out that you can also do destructuring for arrays when assigning variables to different elements to the array. 
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event is anything that can happen from the user's perspective on the webpage, such as clicking on or hover the mouse over an element of the DOM.
```
Comment (andeeisaacs) [inline interview-questions.md:15]: Yessir.

File: interview-questions.md [hunk 11]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer: I forgot to say that DOM events are usually tied to JavaScript functions so that the function is run once the DOM event occurs.
 
 
 
 3. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: The analogy is that a class is like a blueprint, while an object is an instance of that blueprint. A class lays out what key value pairs and methods an object could have, while an object is is an actual entity with real key value pairs and methods to use.
```
Comment (andeeisaacs) [inline interview-questions.md:23]: Totally, my dude. Great answer.

File: interview-questions.md [hunk 12]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 4. What is the difference between a HTML div and a span?
 
-  Your answer:
+  Your answer: I don't really know formally what the difference is. To my understading, a div is a container that can hold many elements, while a span is a container that is meant to only differentiate a piece of text that is in a larger block of text.
 
-  Researched answer:
+  Researched answer: A div is what is called a block level element, while a span is an inline element
```
Comment (andeeisaacs) [inline interview-questions.md:33]: You got the right idea! Great researched answer.

File: interview-questions.md [hunk 13]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 5. What is React.Fragment (or <>) and why would you want to use it?
 
-  Your answer:
+  Your answer: A React.Fragment is similar to a div in that it can hold multiple html elemets inside of it. It is different from a <div> in that it doesn't show nesting such that if you were to pull up the code on the broswer, the individual elements would show as if there was no container. It saves time from having to parse through a bunch of div tags.
 
-  Researched answer:
+  Researched answer: I found out that only if you use the long-hand syntax of a React.Fragment can you assign keys as an attribute to the fragment
```
Comment (andeeisaacs) [inline interview-questions.md:41]: Interesting!

File: interview-questions.md [hunk 14]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 6. What are three options for creating responsive design?
 
-  Your answer:
+  Your answer: Okay, I'm not quite sure what this means, but I think this question is referring to DOM events. You can have code react to an onClick, when your cursor is hovering over an element, or when the element is in frame of the screen.
```
Comment (andeeisaacs) [inline interview-questions.md:47]: Not quite.. but I know you got it right with your researched answer!

File: interview-questions.md [hunk 15]
```diff
@@ -4,56 +4,59 @@ Answer the following questions. First, without external resources. Challenge you
...
+- irb: IRB stands for interactive Ruby. It is a terminal shell that allows you to run Ruby code and see the output, similar ot a terminal.
+- Instance variable: A variable that belongs to an object but not the class iteself.
+- Ruby: An object oriented programming language used to develop web apps. 
+- Class relationships: The four main relationships in object oriented programming are inheritance, association, composition, and aggregation. These are based on a "has-a" , "is-a", or "part-of" relationship.
+- Getter and Setter methods in Ruby: In Ruby, getter and setter methods use "@" to speificy variabels. There are build in methods in Ruby: attr_reader, attr_writer, and attr_accessor to help crate getter and stter methods for you.
```
Comment (andeeisaacs) [inline interview-questions.md:62]: 🥇 🤩 👍

File: react-commenting.js [hunk 16]
```diff
@@ -7,19 +7,19 @@
 class App extends Component{
   constructor(){
     super()
-    // 1.
+    // 1. This defines a state object in the App class
```
Comment (andeeisaacs) [inline react-commenting.js:10]: 👍

File: react-commenting.js [hunk 17]
```diff
@@ -7,19 +7,19 @@
...
     this.state = {
       userInput: "",
     }
   }
 
   robot = (e) => {
-    // 2.
+    // 2. This method is tied to an onChange DOM event specified later in the code. This robot method will change the userInput as the user inputs data into the input section of the page
```
Comment (andeeisaacs) [inline react-commenting.js:17]: You got it, dude.

File: react-commenting.js [hunk 18]
```diff
@@ -7,19 +7,19 @@
...
 
   robot = (e) => {
-    // 2.
+    // 2. This method is tied to an onChange DOM event specified later in the code. This robot method will change the userInput as the user inputs data into the input section of the page 
     this.setState({ userInput: e.target.value })
   }
 
   render(){
-    // 3.
+    // 3. This is a simple destructuring of the userInput variable inside the state object of the App class. This allows for quicker access to the userInput without having to write this.state over and over again.
```
Comment (andeeisaacs) [inline react-commenting.js:22]: Correctamundo.

File: react-commenting.js [hunk 19]
```diff
@@ -28,16 +28,16 @@ class App extends Component{
 
         <h3>Enter text here:</h3>
         <input
-          {/* 4. */}
+          {/* 4. This links the DOM event of onChange to the robot() method. */}
```
Comment (andeeisaacs) [inline react-commenting.js:31]: 🌮

File: react-commenting.js [hunk 20]
```diff
@@ -28,16 +28,16 @@ class App extends Component{
...
           onChange={ this.robot }
-          {/* 5. */}
+          {/* 5. This specifies what the value is for the input tag. this needs to be specified in order for data to be eventually submitted.  */}
```
Comment (andeeisaacs) [inline react-commenting.js:33]: Yup.

File: react-commenting.js [hunk 21]
```diff
@@ -28,16 +28,16 @@ class App extends Component{
...
           value={ userInput }
         />
 
         <div>
-          {/* 6. */}
+          {/* 6. This is calling an instance of the class GoodRobot to be rendered on the page. */}
```
Comment (andeeisaacs) [inline react-commenting.js:38]: Yes. I think another way we can see is that this is the call of a child component as well.

File: react-commenting.js [hunk 22]
```diff
@@ -28,16 +28,16 @@ class App extends Component{
...
           <GoodRobot
-            {/* 7. */}
+            {/* 7. This is passing App's state object property of userInput to be accessible to the GoodRobot class as a prop. */}
```
Comment (andeeisaacs) [inline react-commenting.js:40]: Excellent answer.

File: react-commenting.js [hunk 23]
```diff
@@ -58,14 +58,14 @@ export default App
 class GoodRobot extends Component{
   render(){
     return(
-      // 8.
+      // 8. Render can only have one parent container, which is why everything is wrapped in a div.
```
Comment (andeeisaacs) [inline react-commenting.js:61]: 👍  👍

File: react-commenting.js [hunk 24]
```diff
@@ -58,14 +58,14 @@ export default App
...
       <div>
         <h3>Good Robot</h3>
-        {/* 9. */}
+        {/* 9. Text will be directly rendered on the page. The most updated userInput from state from App.js will be added where specified. The text will be able to be dinamically changed as the user puts text into the input field */}
```
Comment (andeeisaacs) [inline react-commenting.js:64]: 🥇 👍

File: react-commenting.js [hunk 25]
```diff
@@ -58,14 +58,14 @@ export default App
...
         <h4>I hear you saying { this.props.userInput }. Is that correct?</h4>
       </div>
     )
   }
 }
-// 10.
+// 10. All classes need an export. This will allow App.js to access the information in GoodRobot.js.
```
Comment (andeeisaacs) [inline react-commenting.js:70]: Yessir. All components need an export. Great job noticing this last one! (lots of people missed it.)

**Review by andeeisaacs — APPROVED** [review]
TRIP!

OUT.
HERE.
KILLIN.
IT.

You are doing so well. I love your attention to detail and as always, your consideration of edge cases and possible errors. Keep it up. 

MERGE IT UP!

File: code-challenges.js (hunk 1 above)
Comment (jlmooreiii) [reply code-challenges.js:18]: Thanks Andee!!

**Pull request merged**